src/ukify/ukify.py

```diff
@@ -454,16 +436,55 @@ def pairwise(iterable):
     return zip(a, b)
 
 
-def pe_validate(filename):
-    import pefile
+def pe_add_sections(uki: UKI, output):
+    pe = pefile.PE(uki.executable, fast_load=True)
+
+    warnings = pe.get_warnings()
+    if warnings:
+        raise RuntimeError(f'pefile warnings treated as errors: {warnings}')
```

Comment:
Please use a "real" exception, e.g. a custom class like `PEWarningError`.

(`RuntimeException` is used in Java because Java requires all exceptions to be caught, except for `RuntimeException`, so you can't just throw a normal exception without cascading adjustments through all the callers, recursively. But Python doesn't have this, so there is really no reason not to use a normal exception. `RuntimeError` should only be used for "unexpected" exceptions, and never for something that would be reasonably caught and handled by calling code. )